Docs/getting started guide (#689)
input-output-hk/cardano-js-sdk: GETTING_STARTED.md

# Getting started
This is a guide on how to quickly start using the SDK to build your idea, so it is more a guide for clients of the SDK
rather than for contributors to the SDK, although anyone can benefit from it.

> **Warning**
> - The steps below are only suitable for development environments.
> - Postgres username and password are [placeholders](./packages/cardano-services/placeholder-secrets) and should be changed.

## Project Overview
The project is divided into two main parts:

>   1. Provider service, [cardano-services](./packages/cardano-services/README.md): a REST API back-end that can query the Cardano Blockchain and submit transactions.

Reviewer: ```suggestion
  1. [cardano-services](./packages/cardano-services/README.md) consists of several backend applications:
    - *Provider server*: a REST API server that can query Cardano data and submit transactions.
    - *Projector*:  chain follower that projects blocks into PostgreSQL database.
```

Author: [c2bbc7f](https://github.com/input-output-hk/cardano-js-sdk/pull/689/commits/c2bbc7faf956b36b8f1d59cff86343df83506eed)

>   1. Client libraries meant to help with concepts like: key management, cryptography high level helpers, input selection, and many more.

Reviewer: All (besides a couple exceptions like `web-extension`) libraries are suitable for server-side as well as client-side
```suggestion
  1. Libraries meant to help with concepts like: key management, cryptography high level helpers, input selection, and many more.
```

Author: [c2bbc7f](https://github.com/input-output-hk/cardano-js-sdk/pull/689/commits/c2bbc7faf956b36b8f1d59cff86343df83506eed)

          other client libraries.
        - [web-extension](./packages/web-extension/README.md): has the ingredients to build your idea as a web extension.

## Starting the Provider Services on Preprod testnet
This section covers how to get `cardano-services` started on the `preprod` testnet, and begin building your idea or just explore.

Starting `mainnet` or `preview` is a more advanced topic, and is discouraged in the 'Getting Started' phase as it requires additional settings.

### Prerequisites:
  1. Docker compose V2 or later
  1. `cardano-js-sdk` project is cloned locally. No need to install or build.
  1. An x86-64 machine. It will not work on an ARM computer (like Apple M1 chips). This guide is tested on Ubuntu.
  
### Option1: `docker compose`:
  - This option uses `docker compose` to start the bare minimum services.
  - Start Preprod network services:
  ```bash
  $ ./scripts/preprod-network.sh up
  ```
  - Stop Preprod network services:
  ```bash
  $ ./scripts/preprod-network.sh down
  ```
  - Dump the rendered docker compose file:
  ```bash
  $ ./scripts/preprod-network.sh dump
  ```

### Option2: using `yarn`:

> If you are already comfortable with yarn, and have gone through the [System Requirements](#system-requirements), you can run

Reviewer: ```suggestion
If you are already comfortable with yarn, and have gone through the [System Requirements](./README.md#system-requirements), you can run
```

Author: [c2bbc7f](https://github.com/input-output-hk/cardano-js-sdk/pull/689/commits/c2bbc7faf956b36b8f1d59cff86343df83506eed)
